hybrasyl/Objects/User.cs

```diff
@@ -1155,12 +1167,20 @@ internal void UseSpell(byte slot, uint target = 0)
         /// <returns>True or false depending on success.</returns>
         public bool ProcessCastingCost(Castable castable)
         {
-            if (castable.CastCosts?.CastCost == null) return true;
+            if (castable.CastCosts.Count == 0) return true;
+
+            var costStruct = castable.CastCosts.Where(e => e.Class.Contains((Class)Class));

```

Comment:
this variable naming is confusing. This should be simply `var costs` as this is an enumerable.